framework/source/class/qx/ui/table/model/Simple.js

```diff
@@ -62,8 +62,8 @@ qx.Class.define("qx.ui.table.model.Simple",
      */
     _defaultSortComparatorAscending : function(row1, row2)
     {
-      var obj1 = row1[arguments.callee.columnIndex];
-      var obj2 = row2[arguments.callee.columnIndex];
+      var obj1 = row1[qx.ui.table.model.Simple._defaultSortComparatorAscending.columnIndex];
+      var obj2 = row2[qx.ui.table.model.Simple._defaultSortComparatorAscending.columnIndex];
```

Comment:
@johnspackman maybe it's to obvious, but I can't find it: where get's the current column index 
 assigned to `qx.ui.table.model.Simple._defaultSortComparatorAscending.columnIndex`?
I'm having several custom sorting functions which are not defined as static members, but as normal members of classes with tables as children.

Reply:
I'm then using `qx.ui.table.model.Simple.setSortMethods` with the table model instance to assign an arbitrary sorting function for a specific column.

Reply:
It's terrible code. (And unfortunately, possibly even mine, from 10 years ago.)

It's set on line 328, as `comparator.columnIndex = columnIndex;`. So it's setting a property of the function, which changes on each call. Yuck! I had to convince myself that it was actually uniquely setting this for each column. It turns out it is (I'm pretty sure), because it's doing so right in the `sortByColumn` function.

Reply:
this PR is a bit of a quick-fix, i.e. if someone had the time and wanted to reimplement it I'd be happy to withdraw this PR

Reply:
How about someone entirely reimplements Table...

Reply:
@derrell sure, the famous table rewrite ;-) But meanwhile: what about changing the implementation of `qx.ui.table.model.Simple.setSortMethods` by including the column index in a closure:
```javascript
    setSortMethods : function(columnIndex, compare)
    {
      var methods;
      if (qx.lang.Type.isFunction(compare))
      {
        methods =
          {
            ascending  : function(row1, row2)
            {
              return compare(row1, row2, columnIndex);
            },
            descending : function(row1, row2)
            {
              return compare(row2, row1, columnIndex);
            }
          };
      }
      else
      {
        methods =
          {
            ascending  : function(row1, row2)
            {
              return compare.ascending(row1, row2, columnIndex);
            },
            descending : function(row1, row2)
            {
              return compare.descending(row1, row2, columnIndex);
            }
          };
      }
      this.__sortMethods[columnIndex] = methods;
    },
```
and changing the signature of the sorting methods to:
```javascript
    _defaultSortComparatorAscending : function(row1, row2, columnIndex)
    {
      var obj1 = row1[columnIndex];
      var obj2 = row2[columnIndex];
      if (qx.lang.Type.isNumber(obj1) && qx.lang.Type.isNumber(obj2)) {
        var result = isNaN(obj1) ? isNaN(obj2) ?  0 : 1 : isNaN(obj2) ? -1 : null;
        if (result != null) {
          return result;
        }
      }
      return (obj1 > obj2) ? 1 : ((obj1 == obj2) ? 0 : -1);
    },
```